build/build.go

```diff
@@ -531,14 +531,22 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 			return nil, nil, notSupported(nodeDriver, driver.OCIExporter)
 		}
 		if e.Type == "docker" {
-			if len(opt.Platforms) > 1 || len(attests) > 0 {
+			features, err := docker.Features(ctx, e.Attrs["context"])
+			if err != nil {
+				return nil, nil, err
+			}
+			if features[dockerutil.OCIImporter] {
+				// rely on oci importer if available (which supports
+				// multi-platform images), otherwise fall back to docker
+				opt.Exports[i].Type = "oci"
+			} else if len(opt.Platforms) > 1 || len(attests) > 0 {
 				return nil, nil, errors.Errorf("docker exporter does not currently support exporting manifest lists")
```

Comment:
We should turn this into two different error messages:

"does not currently support" - if daemon doesn't support OCI
"does not support, use OCI exporter instead" - if `e.Output != nil`